cpp/src/arrow/compute/key_hash.cc

```diff
@@ -471,7 +473,31 @@ Status Hashing32::HashBatch(const ExecBatch& key_batch, uint32_t* hashes,
 
   LightContext ctx;
   ctx.hardware_flags = hardware_flags;
-  ctx.stack = temp_stack;
+
+  const int64_t alloc_entry_length = column_arrays[0].length();
+  auto estimate_size = [&] {
+    // An estimate TempVectorStack usage size for Hashing32::HashMultiColumm.
+    const int64_t alloc_size1 =
+        2 * (alloc_entry_length * sizeof(uint32_t) + util::TempVectorStack::meta_size());
+    const int64_t alloc_size2 =
+        alloc_entry_length * sizeof(uint16_t) + util::TempVectorStack::meta_size();
+    return alloc_size1 + alloc_size2;
+  };
+
+  if (!temp_stack) {
+    util::TempVectorStack stack;
+    RETURN_NOT_OK(stack.Init(default_memory_pool(), estimate_size()));
+    ctx.stack = std::move(&stack);
```

Comment:
Is this safe?
I think that `stack` is invalid outside of this block.

Reply:
Oh,,, i lost my mind!